### DISCUSSION TITLE
CDN discussion

### DISCUSSION THREAD

**Discussion by dmundra** [uswds/uswds#4679] [Ideas]
## Description

Re-opening a discussion regarding CDN for USWDS assets. Previous issues and comments:

* https://github.com/uswds/uswds/issues/1093
* https://github.com/uswds/uswds/pull/1100#issuecomment-207456638

We are using the USWDS assets with HTML files and it is a lot of files to commit to our repository. A CDN would make that not required and make it easier for other integrations e.g. Drupal https://www.drupal.org/project/uswds

Has there been further discussion about this?

**Comment by aduth** [comment]
One consideration for CDN usage is compatibility of the [Icon component](https://designsystem.digital.gov/components/icon/), which as implemented today _cannot_ work cross-domain, since it relies on `<use href="...">`.

>There is currently no way for a `<use>` element to reuse content from another web domain, even if that content is served with cross-origin HTTP headers.

https://oreillymedia.github.io/Using_SVG/extras/ch10-cors.html

**Reply by mgifford** [reply]
This is for SVG files only, right? https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use

It could we not have most everything served from the CDN, but have a much smaller local file that addresses elements which we know must be managed locally at this point. 

This shouldn't be a barrier, just a bit more work to set up the first time.

**Comment by gavinr** [comment]
I agree that having a CDN solution is useful for quick demos on sites like Codepen and jsbin. Here is one way to use the [UNPKG CDN](https://unpkg.com/) with USWDS:

1. Add CSS via `<link rel="...` tag: https://unpkg.com/uswds@2/dist/css/uswds.min.css
    - e.g. `<link rel="stylesheet" href="https://unpkg.com/uswds@2/dist/css/uswds.min.css" />`
2. Add JS via `<script src="...` tag: https://unpkg.com/uswds@2/dist/js/uswds.min.js
    - e.g. `<script src="https://unpkg.com/uswds@2/dist/js/uswds.min.js"></script>`
3. For a demo, copy template code from https://designsystem.digital.gov/templates/documentation-page/
    - In this template code, find all instances of `/assets/img` and replace with `https://unpkg.com/uswds@2/dist/img`

Unless I'm missing something, the result looks ok to me: 
[![image](https://user-images.githubusercontent.com/209355/177645918-7ebfbf96-5abd-4ad2-a143-7d7eb9703ae4.png)](https://codepen.io/gavinr/pen/oNqbqLw?editors=1000)

https://codepen.io/gavinr/pen/oNqbqLw?editors=1000 ([Use this as a template](https://codepen.io/pen?template=oNqbqLw))
